ibc_public/utils_relaxo.py

```diff
@@ -616,3 +620,200 @@ def t1_pipeline(do_normalise_before=False,
+
+    # data files
+    data_dir = DATA_LOC
+
+    niftis = []
+    jsons = []
+    for fi in listdir(join(data_dir, 'anat')):
+        comps = fi.split('_')
+        if comps[-1] == 'T2star.nii.gz':
+            niftis.append(join(data_dir, 'anat', fi))
+        elif comps[-1] == 'T2star.json':
+            jsons.append(join(data_dir, 'anat', fi))
+        else:
+            continue
+    niftis.sort()
+    jsons.sort()
+
+    # preprocessing directory setup
+    time_elapsed = time.time() - start_time
+    print('[INFO,  t={:.2f}s] copying necessary files...'.format(time_elapsed))
+    cwd = SAVE_TO
+    preproc_dir = join(cwd, 'tmp_t2star', 'preproc')
+    if not exists(preproc_dir):
+        makedirs(preproc_dir)
+
+    # sorting and copying magnitude and phase images to tmp preproc dir
+    mag_niftis = []
+    phase_niftis = []
+    for json_, nifti in zip(jsons, niftis):
+        # copy t2star related files
+        system('cp {} {}'.format(nifti, preproc_dir))
+        nifti = join(preproc_dir, nifti.split(sep)[-1])
+        system('gunzip -df {}'.format(nifti))
+        nifti = join(preproc_dir, nifti.split(sep)[-1].split('.')[0] + '.nii')
+
+        # Load json file as a dictionary
+        json_file = open(json_)
+        json_dict = json.load(json_file)
+
+        # separate phase and magnitude images
+        if 'PHASE' in json_dict['ImageType']:
+            phase_niftis.append(nifti)
+        else:
+            mag_niftis.append(nifti)
+
+    if len(mag_niftis) > 1:
+        return NotImplementedError
+    else:
+        assert len(mag_niftis) == 1
+        mag_nifti = mag_niftis[0]
+        if do_normalise_before:
+            # get segments for better normalisation
+            time_elapsed = time.time() - start_time
+            print('[INFO,  t={:.2f}s] segmenting'.format(time_elapsed))
+            image = mag_nifti
+            out_info = segment(image, True)
+
+            # save normalised segments for later use
+            segments = [join(preproc_dir, f"w{out_info[segment].split('/')[-1]}") for segment in ['gm', 'wm']]
+
+            # normalise image to MNI space
+            time_elapsed = time.time() - start_time
+            print('[INFO,  t={:.2f}s] transforming to MNI space...'.format(time_elapsed))
+            out_info = to_MNI(image, data=out_info)
+            norm_mag_nifti = out_info['anat']
+            
+            time_elapsed = time.time() - start_time
+            print('[INFO,  t={:.2f}s] \t transformed {}'.format(time_elapsed, image.split(sep)[-1]))
+
+        # preprocessing step: segmenting for masking
+        if do_segment:
+            time_elapsed = time.time() - start_time
+            # if already normalised, use MNI mask
+            if do_normalise_before:
+                print('[INFO,  t={:.2f}s] creating a mask'.format(time_elapsed))
+                image = norm_mag_nifti
+                mni = compute_brain_mask(image)
+                closed_mni = closing(mni, 12)
```

Comment:
```suggestion
                closed_mni = closing(mni, closing_param)
```
where `closing_param = 12`